addon/lint/lint.js

```diff
@@ -187,7 +187,7 @@
   CodeMirror.defineOption("lint", false, function(cm, val, old) {
     if (old && old != CodeMirror.Init) {
       clearMarks(cm);
-      if (!state.options.lintOnChange !== false)
+      if (cm.state.lint.options.lintOnChange === undefined || cm.state.lint.options.lintOnChange === true)
```

Comment:
I am not making this up. The (current) following code does not do what it is supposed to do. You can try it yourself. 
```
      if (!state.options.lintOnChange !== false)
        cm.off("change", onChange);
```
I know about precedence of operators. The correct condition is really:
```
cm.state.lint.options.lintOnChange === undefined || cm.state.lint.options.lintOnChange === true
```